src/fileSystem.ts

```diff
@@ -1,276 +1,190 @@
-    older: async () => alertIfPossible(`Sorry, we can't sync your filesystem with this app. Your filesystem version is out-dated and it needs to be migrated.\n\nRun a migration (https://guide.fission.codes/accounts/account-signup/account-migration) or talk to Fission support: support@fission.codes`),
+  // Self delegate file system UCAN & add stuff to cabinet
+  const fileSystemDelegation = await selfDelegateCapabilities(identifier)
+  await cabinet.addUcan(fileSystemDelegation)
+  if (maybeMount) {
+    await cabinet.addAccessKey({
+      key: maybeMount.capsuleRef,
+      path: Path.combine(Path.directory("private"), maybeMount.path),
+    })
   }
-}
 
+  // Add initial CID to log
+  await cidLog.add([
+    await fs.calculateDataRoot(),
+  ])
 
-export async function checkFileSystemVersion(
-  depot: Depot.Implementation,
-  config: Configuration,
-  filesystemCID: CID
-): Promise<void> {
-  const links = await RootTree.linksFromCID(depot, filesystemCID)
-
-  const versionStr = links[ RootBranch.Version ] == null
-    ? "1.0.0"
-    : new TextDecoder().decode(
-      await DAG.getRaw(depot, links[ RootBranch.Version ])
-    )
-
-  const errorVersionBigger = async () => {
-    await (config.userMessages || DEFAULT_USER_MESSAGES).versionMismatch.newer(versionStr)
-    return new Error(`Incompatible filesystem version. Version: ${versionStr} Supported versions: ${Versions.supported.map(v => Versions.toString(v)).join(", ")}. Please upgrade this app's ODD SDK version.`)
-  }
-
-  const errorVersionSmaller = async () => {
-    await (config.userMessages || DEFAULT_USER_MESSAGES).versionMismatch.older(versionStr)
-    return new Error(`Incompatible filesystem version. Version: ${versionStr} Supported versions: ${Versions.supported.map(v => Versions.toString(v)).join(", ")}. The user should migrate their filesystem.`)
-  }
-
-  const versionParsed = Versions.fromString(versionStr)
-
-  if (versionParsed == null) {
-    throw await errorVersionBigger()
-  }
-
-  const support = Versions.isSupported(versionParsed)
-
-  if (support === "too-high") {
-    throw await errorVersionBigger()
-  }
-  if (support === "too-low") {
-    throw await errorVersionSmaller()
-  }
+  // Fin
+  return fs
 }
 
-
-function alertIfPossible(str: string) {
-  if (globalThis.alert != null) globalThis.alert(str)
+/**
+ * Fetches the account DID.
+ */
+export async function accountDID<Annex extends AnnexParentType>({ account, cabinet, identifier }: {
+  account: Account.Implementation<Annex>
+  cabinet: Cabinet
+  identifier: Identifier.Implementation
+}): Promise<string> {
+  const identifierDID = await identifier.did()
+  const identifierUcans = cabinet.audienceUcans(identifierDID)
+  return account.did(identifierUcans, cabinet.ucansIndexedByCID)
 }
```

Comment:
Writing this out to check my understanding of the interaction between the identifier and the account system.

The user will have a signing key which is associated with their identifier DID:

```typescript
const identifierDID = await identifier.did()
```

The account system will have delegated authority to the identifier DID, giving it the ability to sync files to the service (e.g. the Fission server). We gather any UCANs that have delegated to the identifier DID:

```typescript
const identifierUcans = cabinet.audienceUcans(identifierDID)
```

Once we have these UCANs, we can determine the account DID (the issuer in one of the UCANs):

```typescript
return account.did(identifierUcans, cabinet.ucansIndexedByCID)
```  

The account DID is associated with a key pair created on the Fission server at account creation time, which is immediately burned to prevent the server from mutating the user's file system without permission from the user.

Am I understanding that all correctly?

Reply:
> Am I understanding that all correctly?

Pretty much yeah.

> The account DID is associated with a key pair created on the Fission server at account creation time, which is immediately burned to prevent the server from mutating the user's file system without permission from the user.

I just realised made some mistakes here. Blaine very recently made this diagram: https://github.com/fission-codes/fission-server/blob/main/fission-server/docs/account-creation.md

The account DID isn't going to be the identifier DID of the device that you used to sign up. Instead the public key of the key pair generated on the server. So instead of looking for the root issuer, we'll have to look for the UCAN we got back from the server containing the server capabilities.

Reply:
Alright, thanks. The pieces are all starting to make sense now. 😄